lib/src/types/binary_codec.dart

```diff
@@ -807,7 +815,15 @@ class PostgresBinaryDecoder {
         if (dinput.timeZone.forceDecodeTimestampAsUTC) {
           return DateTime.utc(2000).add(Duration(microseconds: value));
         }
-        return DateTime(2000).add(Duration(microseconds: value));
+        // https://github.com/dart-lang/sdk/issues/56312
+        // ignore previous timestamp transitions and use only the current system timestamp in local date and time so that the behavior is correct on Windows and Linux
+        final nowDt = DateTime.now();
```

Comment:
Let's extract these two into a method that does the correction.

Reply:
yes definitely